src/lib/tsc-watch.ts

```diff
@@ -71,6 +79,14 @@ function killProcesses(currentCompilationId: number, killAll: boolean): Promise<
   return runningKillProcessesPromise;
 }
 
+function debounce<T extends (...args: Parameters<T>) => void>(this: ThisParameterType<T>, fn: T, delay = 300) {
+  let timer: ReturnType<typeof setTimeout> | undefined
+  return (...args: Parameters<T>) => {
+    timer && clearTimeout(timer)
+    timer = setTimeout(() => fn.apply(this, args), delay)
+  }
+}
+
```

Comment:
This is some utility function that should be on some other file, like debounce.ts